### PR TITLE
fix: Fixes color mapping issues by adding the ColorMap property 

### PR DESCRIPTION
**Pull request by Golodhros**
## Description
Adds the colorMap property to all charts that need it
Updates types
Makes 'isAnimated' false by default in all charts
Removes uses of dataLabel, valueLabel, and other data labeling functions

## Motivation and Context
Fixes https://github.com/britecharts/britecharts/issues/431 and adds new possibilities to color configurations.
Part of https://github.com/britecharts/britecharts/issues/870

## How Has This Been Tested?
Added tests

## Types of changes

-   [ ] Refactor (changes the way we code something without changing its functionality)
-   [x] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] Latest master code has been merged into this branch
-   [ ] No commented out code (if required, place // TODO above with explanation)
-   [ ] No linting issues
-   [ ] Build is successful
-   [ ] Code follows the [API Guidelines](http://britecharts.github.io/britecharts/topics-index.html#toc5__anchor)
-   [ ] Updated the documentation
-   [ ] Added tests to cover changes
-   [ ] All new and existing tests passed

### CODE REVIEW

File: demos/src/demo-stacked-bar.js [hunk 1]
```diff
@@ -35,9 +35,6 @@ function createStackedBarChartWithTooltip(optionalColorSchema) {
             .width(containerWidth)
             .grid('horizontal')
             .isAnimated(true)
-            .stackLabel('stack')
```
Comment (DalerAsrorov) [inline demos/src/demo-stacked-bar.js:38]: How does this change the rendering of the chart?

File: src/charts/bar.js [hunk 2]
```diff
@@ -326,8 +325,17 @@ export default function module() {
...
-        colorMap = (item) =>
-            colorList.filter(({ name }) => name === item)[0].color;
+        nameToColorMap =
+            nameToColorMap ||
+            data
+                .map((d) => d)
+                .reduce(
+                    (acc, { name }, i) => ({
+                        ...acc,
+                        [name]: colorSchema[i % colorSchema.length],
```
Comment (DalerAsrorov) [inline src/charts/bar.js:335]: what does the modulo operator do in this case?

File: src/typings/charts/line-chart.d.ts [hunk 3]
```diff
@@ -1,109 +1,113 @@
...
+    TopicName = 'topicName',
+    Value = 'value',
 }
 
 export type LineChartDataShape = {
-  [LineChartKeys.Date]: string;
-  [LineChartKeys.Name]: string;
-  [LineChartKeys.TopicName]: string;
-  [LineChartKeys.Value]: number;
+    [LineChartKeys.Date]: string;
```
Comment (DalerAsrorov) [inline src/typings/charts/line-chart.d.ts:19]: is the formatting still preserved? I assume it follows the eslint rules or Prettier configs we set

**Review by DalerAsrorov — APPROVED** [review]
few comments but looks great!

File: demos/src/demo-stacked-bar.js (hunk 1 above)
Comment (Golodhros) [reply demos/src/demo-stacked-bar.js:38]: It doesn't!

File: src/charts/bar.js (hunk 2 above)
Comment (Golodhros) [reply src/charts/bar.js:335]: It applies one color from the color schema to each category, starting again when it reaches the max number of elements, so for a colorSchema of 2 colors:
- first element, i = 0, color colorSchema[0]
- second element, i = 1, color colorSchema[1]
- third element, i = 2, color colorSchema[0]
and so on

File: src/typings/charts/line-chart.d.ts (hunk 3 above)
Comment (Golodhros) [reply src/typings/charts/line-chart.d.ts:19]: Prettier, yes.
We only lose the `.append` thing, but that gave more work than what was worth it!

**Pull request merged**